pkg/networkservice/common/discover/server_test.go

```diff
@@ -187,7 +187,7 @@ func TestMatchEmptySourceSelector(t *testing.T) {
 		checkcontext.NewServer(t, func(t *testing.T, ctx context.Context) {
 			nses := discover.Candidates(ctx).Endpoints
 			require.Len(t, nses, 1)
-			require.Equal(t, want, nses[0].NetworkServiceLabels)
+			require.Equal(t, want, nses[0][0].NetworkServiceLabels)
```

Comment:
Please, add additional check: `require.Len(t, nses[0], 1)`
Same for other tests with such checks.